run.ps1

```diff
@@ -0,0 +1,30 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path $PSScriptRoot ".tools"
+$bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+
+# if the boot-strapper script doesn't exist then download it
+if ((Test-Path $bootStrapperPath) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+}
+
+# now execute it
+Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+if ($LastExitCode -ne 0)
+{
+    Write-Output "Boot-strapping failed with exit code $LastExitCode"
+    exit $LastExitCode
+}
+
+# execute the tool using the dotnet.exe host
+$dotNetExe = Join-Path $toolsLocalPath "dotnetcli\shared\Microsoft.NETCore.App\version\dotnet.exe"
```

Comment:
The problem here is that we cannot use the dotnet.exe that's under the version directory to run "dotnet restore" as it can't find the restore command (it's implemented in some other dll).  We can either hack up the CLI/FX installation to resolve this or we bifurcate the implementation to have "dotnet restore" use the "top-level" dotnet.exe and use the version-specific dotnet.exe purely as a host (I don't like the idea of hacking up the installation).  Perhaps there are other options here too.